fault/tester/staged_tester.py

```diff
@@ -127,71 +113,25 @@ def make_target(self, target: str, **kwargs):
             return SpiceTarget(self._circuit, **kwargs)
         raise NotImplementedError(target)
 
-    def is_recursive_type(self, T):
-        return issubclass(T, m.Tuple) or issubclass(T, m.Array) and \
-            not issubclass(T.T, m.Digital)
-
-    def get_type(self, port):
-        if isinstance(port, SelectPath):
-            port = port[-1]
-        if isinstance(port, WrappedVerilogInternalPort):
-            type_ = port.type_
-            print(port.type_)
-        else:
-            type_ = type(port)
-        return type_
-
     def poke(self, port, value, delay=None):
         """
         Set `port` to be `value`
         """
-        # set defaults
-        if delay is None:
-            delay = self.poke_delay_default
-
-        if port is self.clock:
-            self.clock_initialized = True
-
-        def recurse(port):
-            if isinstance(value, dict):
-                for k, v in value.items():
-                    self.poke(getattr(port, k), v)
-            elif isinstance(port, m.Array) and \
-                    not issubclass(type(port).T, m.Digital) and \
-                    isinstance(value, (int, BitVector, tuple, dict)):
-                # Broadcast value to children
-                for p in port:
-                    self.poke(p, value, delay)
-            else:
-                _value = value
-                if isinstance(_value, int):
-                    _value = BitVector[len(port)](_value)
-                for p, v in zip(port, _value):
-                    self.poke(p, v, delay)
-
-        # implement poke
-        if isinstance(port, SelectPath):
-            if (self.is_recursive_type(type(port[-1]))
-                or (not isinstance(port[-1], WrappedVerilogInternalPort) and
-                    isinstance(port[-1].name, m.ref.AnonRef))):
-                return recurse(port[-1])
-        elif self.is_recursive_type(type(port)):
-            return recurse(port)
-        elif not isinstance(port, WrappedVerilogInternalPort) and\
-                isinstance(port.name, m.ref.AnonRef):
-            return recurse(port)
+        recursed = super().poke(port, value, delay)
+        if recursed:
+            return
```

Comment:
should this also return a `bool` so that `Tester::poke` and `AbstractTester::poke` have the same interface? i.e. `return True` here and `return False` at end of func.

Reply:
It's not meant to be part of the external interface, it's just to reuse the recursive logic provided by the base class.

Reply:
Got it. In that case would recommend doing the following:

In `TesterBase`:

```python
def _poke(...):
    # Whatever logic was originally in poke() before.

def poke(...):
    self._poke(...)
```

Then in this class `Tester` override `poke()` and refer to `super()._poke(...)`. That way `poke()` has the same interface across and `super._poke()` can be reused.

Not a big deal either way, just seems like a cleaner/more consistent interface.